src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/IncrementalBuildTest.cs

```diff
@@ -516,6 +516,68 @@ public void ProduceReferenceAssembly ()
 			}
 		}
 
+		[Test]
+		public void TransitiveDependencyProduceReferenceAssembly ()
+		{
+			var path = Path.Combine (Root, "temp", TestName);
+			var app = new XamarinAndroidApplicationProject {
+				ProjectName = "App",
+				Sources = {
+					new BuildItem.Source ("Class1.cs") {
+						TextContent = () => "public class Class1 : Library1.Class1 { }"
+					},
+				}
+			};
+			var lib1 = new DotNetStandard {
+				ProjectName = "Library1",
+				Sdk = "Microsoft.NET.Sdk",
+				TargetFramework = "netstandard2.0",
+				Sources = {
+					new BuildItem.Source ("Class1.cs") {
+						TextContent = () => "namespace Library1 { public class Class1 { } }"
+					},
+					new BuildItem.Source ("Class2.cs") {
+						TextContent = () => "namespace Library1 { public class Class2 : Library2.Class1 { } }"
+					}
+				}
+			};
+			lib1.SetProperty ("ProduceReferenceAssembly", "True");
+			var lib2 = new DotNetStandard {
+				ProjectName = "Library2",
+				Sdk = "Microsoft.NET.Sdk",
+				TargetFramework = "netstandard2.0",
+				Sources = {
+					new BuildItem.Source ("Class1.cs") {
+						TextContent = () => "namespace Library2 { public class Class1 { } }"
+					},
+				}
+			};
+			lib2.SetProperty ("ProduceReferenceAssembly", "True");
+			lib1.OtherBuildItems.Add (new BuildItem.ProjectReference ($"..\\{lib2.ProjectName}\\{lib2.ProjectName}.csproj", lib2.ProjectName, lib2.ProjectGuid));
```

Comment:
Why does this use `lib1.OtherBuildItems` and not `lib1.References`?

Reply:
Oh yeah, that's a little tricky thing I learned when setting up this test. What happens is that `XmlUtils.ToXml()` _only_ adds `OtherBuildItems` in projects that have `$(EnableDefaultItems)`=`true`:

https://github.com/xamarin/xamarin-android/blob/99ec9f231bc52b4e9c451b955f2493517afc1fc2/src/Xamarin.Android.Build.Tasks/Tests/Xamarin.ProjectTools/Utilities/XmlUtils.cs#L32-L51

And `DotNetStandard` has `$(EnableDefaultItems)`=`true`:

https://github.com/xamarin/xamarin-android/blob/99ec9f231bc52b4e9c451b955f2493517afc1fc2/src/Xamarin.Android.Build.Tasks/Tests/Xamarin.ProjectTools/Common/DotNetStandard.cs#L42